soc/arm/quicklogic_eos_s3/CMakeLists.txt

```diff
@@ -5,4 +5,5 @@ zephyr_include_directories(${ZEPHYR_BASE}/drivers)
 zephyr_sources(
   soc.c
   irq_handlers.c
+  fpga_loader.c
```

Comment:
`fpga_loader.c` should be added to compilation only when we want to use FPGA.
```suggestion
```
Please add `zephyr_sources_ifdef(CONFIG_EOS_S3_PROGRAM_FPGA fpga_loader.c` below